internal/cluster/distribution/pke/node_pool.go

```diff
@@ -0,0 +1,40 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pke
+
+import (
+	"context"
+)
+
+type ExistingNodePool struct {
+	Name string
+}
+
+// +testify:mock
+
+// NodePoolStore provides an interface for PKE node pool persistence.
+type NodePoolStore interface {
+	// DeleteNodePool deletes an existing node pool from the storage.
+	DeleteNodePool(ctx context.Context, organizationID, clusterID uint, clusterName string, nodePoolName string) error
+
+	// ListNodePools retrieves the node pools for the cluster specified by its
+	// cluster ID.
+	ListNodePools(
```

Comment:
As far as I can tell, this only exists (together with `ExistingNodePool`), because the list activity calls it. But I don't think the list activity is used anywhere, so this should be dropped with it.

Reply:
It's called from the service as well.In the delete [function](https://github.com/banzaicloud/pipeline/pull/3242/files#diff-d746cfc80d0e748007eb394ac902c14cedde894d4b3e652c2e7dcc10d787290aR163)